notifications_utils/clients/redis/__init__.py

```diff
@@ -3,8 +3,11 @@
 from .request_cache import RequestCache  # noqa: F401 (unused import)
 
 
-def daily_limit_cache_key(service_id):
-    return f"{service_id}-{datetime.utcnow().strftime('%Y-%m-%d')}-count"
+def daily_limit_cache_key(service_id, notification_type=None):
+    if not notification_type:
+        return f"{service_id}-{datetime.utcnow().strftime('%Y-%m-%d')}-count"
+
+    return f"{service_id}-{notification_type}-{datetime.utcnow().strftime('%Y-%m-%d')}-count"
```

Comment:
I was tempted to do this refactor when I moved this to using f-strings. Now that we’re repeating it I think it’s definitely worth refactoring into something a bit more readable:
```suggestion
    yyyy_mm_dd = datetime.utcnow().strftime('%Y-%m-%d')
    
    if not notification_type:
        return f"{service_id}-{yyyy_mm_dd}-count"

    return f"{service_id}-{notification_type}-{yyyy_mm_dd}-count"
```